regolith/helpers/l_generalhelper.py

```diff
@@ -0,0 +1,84 @@
+"""Helper for finding people's information in the people collection
+"""
+
+from regolith.helpers.basehelper import SoutHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import (
+    all_docs_from_collection,
+    get_pi_id,
+    search_collection,
+    collection_str,
+)
+HELPER_TARGET = "lister"
+
+def subparser(subpi):
+    subpi.add_argument(
+        "coll",
+        help='collection where the lister is going to run.')
+    subpi.add_argument(
+        "--kv_filter",
+        nargs="+",
+        help='search the given collection by key element pairs')
+    subpi.add_argument(
+        "-r",
+        nargs="+",
+        help='the name of the fields to return from the search')
+    subpi.add_argument(
+        "-k", "--keys",
+        action="store_true",
+        help='list of the available keys and their description')
+    return subpi
+
+class GeneralListerHelper(SoutHelperBase):
+    """Helper for finding and listing contacts from the contacts.yml file
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = ["abstracts", "assignments","beamplan", "beamtime",
```

Comment:
this is not maintainable, and will make this function very slow.  Maybe it can work if you put the variable assignment into the ``construct_global_ctx`` method and define it as:
```
needed_dbs = [rc.coll]
```

Reply:
done!